src/main/java/com/hm/achievement/utils/AchievementPlaceholderHook.java

```diff
@@ -25,28 +26,62 @@ public AchievementPlaceholderHook(AdvancedAchievements plugin) {
 
 	@Override
 	public String onPlaceholderRequest(Player p, String identifier) {
+
+		int achievements = plugin.getCacheManager().getPlayerTotalAchievements(p.getUniqueId());
```

Comment:
This call may be expensive if the value is not cached. Therefore I suggest moving this down into the `if` conditions to make sure it's only computed when needed. ^^

Reply:
Oups! Haha yes... ^^'